.github/workflows/build_and_test.yml

```diff
@@ -761,6 +761,9 @@ jobs:
       run: ./dev/lint-r
     - name: Run documentation build
       run: |
+        # Build docs first with SKIP_API to ensure they are buildable without requiring any
+        # language docs to be built beforehand.
+        cd docs; SKIP_API=1 bundle exec jekyll build; cd ..
         if [ -f "./dev/is-changed.py" ]; then
           # Skip PySpark and SparkR docs while keeping Scala/Java/SQL docs
           pyspark_modules=`cd dev && python3.9 -c "import sparktestsupport.modules as m; print(','.join(m.name for m in m.all_modules if m.name.startswith('pyspark')))"`
```

Comment:
or wonder if we should set `SKIP_SCALADOC=1` together with `if [ `./dev/is-changed.py -m ...` condition. That might be better.

Reply:
Hmm, could you elaborate a little bit on this? How would this prevent [yesterday's scenario][1]?

[1]: https://github.com/apache/spark/pull/44393#discussion_r1444169083

Reply:
Actually let's just merge this. 5 secs won't affect the elapsed time much anyway.